astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -1555,11 +1573,8 @@ def __init__(
         :type name: str or None
         """
 
-        self.doc = doc
-        """The function's docstring.
-
-        :type doc: str or None
-        """
+        self._doc = doc
```

Comment:
```suggestion
        self._doc = None
        if doc is not None:
            warnings.warn("Specific warning for the constructor ?") 
            self.doc = doc
```
Would be the occasion to tell the user what he should do instead (use postinit ?). We can also use the property directly instead of assigning to ``_doc``.

Reply:
I think that is step 4 of @cdce8p's deprecation plan. For now we only want to warn when users access `node.doc`.